src/combinatorics/set_family/kleitman.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import combinatorics.set_family.harris_kleitman
+import combinatorics.set_family.intersecting
+
+/-!
+# Kleitman's bound on the size of intersecting families
+
+An intersecting family on `n` elements has size at most `2ⁿ⁻¹`, so we could naïvely think that two
+intersecting families could cover all `2ⁿ` sets. But actually that's not case because for example
+none of them can contain the empty set. Intersecting families are in some sense correlated.
+Kleitman's bound stipulates that `k` intersecting families cover at most `2ⁿ - 2ⁿ⁻ᵏ` sets.
+
+Note that, as opposed to the one family case, not all maximal unions of `k` families have this size.
+
+## Main declarations
+
+* `finset.card_bUnion_le_of_intersecting`: Kleitman's theorem.
+
+## References
+
+* [D. J. Kleitman, *Families of non-disjoint subsets*][kleitman1966]
+-/
+
+open finset fintype (card)
+
+variables {ι α : Type*} [fintype α] [decidable_eq α] [nonempty α]
+
+/-- **Kleitman's theorem**. An intersecting family on `n` elements contains at most `2ⁿ⁻¹` sets, and
+each further intersecting family takes at most half of the sets that are in no previous family. -/
+lemma finset.card_bUnion_le_of_intersecting (s : finset ι) (ℱ : ι → finset (finset α))
```

Comment:
I don't really think we should be using script letters for set families - I think it's more difficult to read and doesn't really add value. The change doesn't need to be in this PR though.

Reply:
For indexed families I agree. But for set families I need to distinguish between terms of type `α`, `finset α` and `finset (finset α)` and script letters fulfill the third role perfectly.